src/accordion/section.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import uniqueId from '../utils/unique-id.js';
+import styles from './section.css';
+
+const headings = {
+  1: 'h1',
+  2: 'h2',
+  3: 'h3',
+  4: 'h4',
+  5: 'h5',
+  6: 'h6',
+};
+
+export default class Section extends React.Component {
+  constructor() {
+    super();
+    this.accessibleId = uniqueId();
+  }
+
+  render() {
+    const {
+      cannotClose,
+      children,
+      cssContainer,
+      cssHeading,
+      cssPanel,
+      cssTrigger,
+      headingLevel,
+      onClick,
+      open,
+      title,
+    } = this.props;
+    const titleVal = typeof title === 'function' ? title({ open }) : title;
+    const Heading = headings[headingLevel];
+
+    if (!Heading) {
+      throw new Error('Accordion must have a headingLevel between 1 and 6');
+    }
+
+    return (
+      <div className={cssContainer}>
+        <Heading className={cssHeading}>
+          <button
+            aria-controls={this.accessibleId}
+            aria-disabled={cannotClose}
+            aria-expanded={open}
+            className={cssTrigger}
+            type="button"
+            onClick={onClick}
+          >
+            {titleVal}
+          </button>
+        </Heading>
+        <div className={open ? cssPanel : undefined} id={this.accessibleId}>
+          {open && children}
+        </div>
+      </div>
+    );
+  }
+}
+
+Section.propTypes = {
+  cannotClose: PropTypes.bool,
```

Comment:
`cannotClose` is an interesting one, and maybe there's a better way to word or implement it.

Some accordions must have at least panel open (ie - you can't close all of them). In that case, the one that's open must have `aria-disabed="true"` set on its "button".

Reply:
I think `cannotClose` is fine. Alternatively we could call it `disabled` and/or also have it add the disabled attribute to the element